MIPs/mip-x.md

```diff
@@ -0,0 +1,129 @@
+
+**Invoker**: This refers to the entity that initiates or "calls" a specific method or function. In the case of `wallet_revokePermissions`, the invoker might be the user or the user's wallet software that initiates the revocation of permissions for specific dApps.
+
+
+## Proposal Specification
+The `wallet_revokePermissions` method is proposed as a new JSON-RPC feature for MetaMask, aimed at giving users more granular control over permission management. With this method, users can either revoke all permissions associated with connected origins (dApps) or opt for a more targeted approach by specifying the target permission. For example, MetaMask can revoke permissions for a specific account address or for many addresses. Additionally, if all parameters are omitted, it triggers a full revocation of all permissions.
+
+The updated method signature will be as follows:
+
+```
+await window.ethereum.request({
+  "method": "wallet_revokePermissions",
+  "params": {
+    "permission": {
+      "caveatValue": string[], // value of the caveat to be revoked from the target permission
+      "target": "eth_accounts" | "wallet_snap" | "snap_dialog" | "snap_notify" | "snap_manageState"
+      "caveatType": string // type of the caveat to update,
+      "id": string // id of permission to be revoked
+    }
+  }
+});
+```
+
+The proposed changes have been implemented in the following PR against the `MetaMask/api-specs` repo: https://github.com/MetaMask/api-specs/pull/145
+
+## Caveats
+The implementation of `wallet_revokePermissions` means more granular control over user permissions and, subsequently, more combinations of permissions that can be revoked. This increases the scope of testing to ensure that there are no edge cases that haven't been considered, or bugs.
+
+## Implementation
+The MetaMask team will be responsible for implementing the proposed changes to the `wallet_revokePermissions` method.
+
+## Developer Adoption Considerations
+1. Backward Compatibility: Dapps that currently manage permissions using custom logic will need to update their code to integrate this new method, however the method is backwards compatible.
+
+2. Ease of Adoption: This method has been designed with flexibility in mind, offering both broad and specific options for permission revocation. This dual capability greatly simplifies the adoption process for both users and dApp developers.
+
+## User Experience Considerations
+
+#### Enhancements:
+
+Streamlining Disconnection: Reducing the number of clicks and steps needed to disconnect from a dApp, making the experience more user-friendly.
+
+Consistency in Connection Management: Providing a disconnect feature directly within the dApp aligns with user expectations and creates a consistent experience.
```

Comment:
can you elaborate on what you mean by consistent here? Consistent that the connect and disconnect can both be requested by the dapp?

Reply:
Re: UX. Like on connect to a dapp, I would expect to have MetaMask notify me of disconnect. I would not necessarily trust that a dapp has disconnected itself just from the dapp ui itself